src/main/java/org/apache/sysds/runtime/matrix/data/LibMatrixDatagen.java

```diff
@@ -590,13 +602,26 @@ private static void genDense(MatrixBlock out, int clen, int blockrows, int block
 	}
 	
 	private static void genFullyDense(DenseBlock c, int blockrows, int blockcols, int rowoffset, int coloffset, 
-		double min, double range, PRNGenerator valuePRNG)
+		double min, double range, IPRNGenerator valuePRNG, long[] ctr)
 	{
-		for(int i = rowoffset; i < rowoffset+blockrows; i++) {
-			double[] cvals = c.values(i);
-			int cix = c.pos(i, coloffset);
-			for(int j = 0; j < blockcols; j++)
-				cvals[cix+j] = min + (range * valuePRNG.nextDouble());
+		if (valuePRNG instanceof PRNGenerator) {
+			for(int i = rowoffset; i < rowoffset+blockrows; i++) {
```

Comment:
please split it into two different functions for each branch of this if statement

Reply:
I changed it to a solution using iterators. This removes the duplicate code. What do you think?